src/shared/user-record-show.c

```diff
@@ -124,13 +124,25 @@ void user_record_show(UserRecord *hr, bool show_full_group_info) {
         case -ESTALE:
                 printf(" Password OK: %slast password change in future%s\n", ansi_highlight_yellow(), ansi_normal());
                 break;
-
         default:
                 if (r < 0) {
                         errno = -r;
                         printf(" Password OK: %sno%s (%m)\n", ansi_highlight_yellow(), ansi_normal());
                         break;
                 }
+                if (!hr->hashed_password) {
+                        printf(" Password OK: %sno%s (empty)\n",
+                               user_record_disposition(hr) == USER_REGULAR ? ansi_highlight_yellow() :
+                                                                             ansi_highlight_red(),
+                               ansi_normal());
+                        break;
+                }
+                if (strv_isempty(hr->hashed_password) || *hr->hashed_password[0] == 33 ||
+                    *hr->hashed_password[0] == 42) {
```

Comment:
hmm, i wonder if we shouldn't rather check if the first char is `!= '$'` since all unix password hashes start with a `$`.

we should probably have a helper function in src/basic/user-util.h:

```c
static inline bool hashed_password_is_locked_or_invalid(const char *password) {
        return password && password[0] == '$';
}
```

and then use that here.

Given that we allow multiple passwords to be set for a user record (unlike traditional NSS) we should iterate through all hashed passwords and only claim the password was locked if all of them appear locked.

Reply:
Does that mean there can be several hashes in `hr->hashed_password` ? 
If yes, are they NULL terminated or anything like that?